### PR TITLE
Refactored UDTs into separate header and cpp files.

### PR DESCRIPTION
**Pull request by JerryLeeMelton**

### CODE REVIEW

File: MusicStoreFastFoodCombo.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <iostream>
```
Comment (matkatmusic) [inline MusicStoreFastFoodCombo.h:3]: you are not using anything from iostream in this header file
- [ ] revise

File: PetStore.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <iostream>
```
Comment (matkatmusic) [inline PetStore.h:3]: you are not using anything from iostream in this header file
- [ ] revise

File: FastFoodRestaurant.cpp [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    chameleon.changeColor(150, 75, 0);
+    return true;
+}
+
+void FastFoodRestaurant::calculateProfit()
+{
+    while(burgersSold > 0)
+    {
+        totalProfits += profitPerBurger;
+        burgersSold -= 1;
```
Comment (matkatmusic) [inline FastFoodRestaurant.cpp:67]: use pre-decrement instead of `-= 1;`
- [ ] revise

File: FastFoodRestaurant.cpp [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    while(potatoesSold > 0)
+    {
+        totalProfits += profitPerPotato;
+        potatoesSold -= 1;
```
Comment (matkatmusic) [inline FastFoodRestaurant.cpp:73]: use pre-decrement instead of `-= 1;`
- [ ] revise

File: Chameleon.cpp [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    red = newRedValue;
+    green = newGreenValue;
+    blue = newBlueValue;
+}
+
+void Chameleon::Color::setAlpha(int newAlphaValue)
+        {
+            alpha = newAlphaValue;
+        }
+
```
Comment (matkatmusic) [inline Chameleon.cpp:24]: formatting
- [ ] revise

**Comment by JerryLeeMelton** [comment]
K, I made revisions based on your notes.  I do have a question though about the comment you made on FastFoodRestaurant.h though. 

You mentioned that I should be using forward declaration for the Chameleon struct instead of including the Chameleon.h header file, but when I tried replacing the include with a forward declaration I was suddenly getting an error from line 58 in FastFoodRestaurant.cpp ("member access into incomplete type 'Chameleon'").

I couldn't quite figure out what I was doing wrong.  Could you maybe point me in the right direction?

File: FastFoodRestaurant.h [hunk 6]
```diff
@@ -1,9 +1,10 @@
 #pragma once
 
-#include <iostream>
 #include "LeakedObjectDetector.h"
 #include "Chameleon.h"
```
Comment (matkatmusic) [inline FastFoodRestaurant.h:4]: Forward declare in the header file.  Include in the cpp file
- [ ] revise

**Pull request merged**